shared/base_component.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import StateManager from 'shared/state_manager'
+import Router from 'shared/router'
+
+export default class BaseComponent extends React.Component {
+	constructor(props, context) {
+ 		super(props, context);
+  		this.state = {};
+ 	}
+	//Must
+	get state_manager(){
+    	return this.context.state_manager;
+  	}
+  	get i18n(){
+    	return this.context.i18n;
+  	}
+  	get router() {
+  		return this.context.router;
+  	}
+
+	get current_route(){
+		return this.router.routes.getRoute(this.route_name);
+	}
+
+	get route_name() {
+		return this.props.location.get('route_name');
+	}
+	pushRoute(route_name, action, payload){
+		this.router.pushRoute(route_name, action, payload);
+	}
+
+	get logged_in(){
+		if(!this.props.session){
+			return false;
+		}
+      	return !!this.props.session.get('token');
+    }
+    
+    get authenticable() {
+    	return false;
+    }
+
+    render(){
+      let component = this;
+      if (component.logged_in || !component.authenticable){
+        return component.template.call(component);
+      } else {
+        return React.createElement(Login, {
+          login: this.props.login
+        });
+      }
+    }
+	get moment() {
+		let i18n = this.context.i18n;
+		if(i18n && i18n.language) {
+			moment.locale(i18n.language);
+		}
+		return moment;
```

Comment:
is this `momentjs`. If so, let's take it out and all client to implement in application repo.